src/react-components/home-root.js

```diff
@@ -35,9 +35,39 @@ class HomeRoot extends Component {
   componentDidMount() {
     this.loadEnvironments();
     this.setState({ dialogType: this.props.dialogType });
-    document.querySelector("#background-video").playbackRate = 0.75;
+    this.loadHomeVideo();
   }
 
+  loadHomeVideo = () => {
+    const videoEl = document.querySelector("#background-video");
+    function initVideo() {
+      videoEl.playbackRate = 0.75;
+      videoEl.play();
+      let inIframe = false;
+      try {
+        inIframe = !!window.frameElement || window.self !== window.top;
+      } catch (e) {
+        inIframe = false;
+      }
+      function toggleVideo() {
+        // Play the video if the window/tab is visible (or is within an `<iframe>`).
+        if (document.hasFocus() || inIframe) {
+          videoEl.play();
+        } else {
+          videoEl.pause();
+        }
+      }
+      document.addEventListener("visibilitychange", toggleVideo);
+      window.addEventListener("focus", toggleVideo);
+      window.addEventListener("blur", toggleVideo);
+    }
+    if (videoEl.readyState >= videoEl.HAVE_FUTURE_DATA) {
```

Comment:
I just have to take a second to lol at these state properties of [`HTMLMediaElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState#Syntax)